src/openlcb/AliasAllocator.hxx

```diff
@@ -126,26 +126,46 @@ public:
      * is not in the alias cache yet.*/
     NodeAlias get_new_seed();
 
-    /** "Allocate" a buffer from this pool (but without initialization) in
-     * order to get a reserved alias. */
-    QAsync *reserved_aliases()
-    {
-        return &reserved_alias_pool_;
-    }
+    /** Allocates an alias from the reserved but unused aliases list. If there
+     * is a free alias there, that alias will be reassigned to destination_id
+     * in the local alias cache, and done will never be notified. If there is
+     * no free alias, then a new alias will be allocated, and done will be
+     * notified when the allocation is complete. Then the call has to be
+     * re-tried by the destination flow.
+     * @param destination_id if there is a free alias right now, it will be
+     * assigned to this Node ID in the local alias cache.
+     * @param done if an async allocation is necessary, this will be notified
+     * after a new alias has been received.
+     * @return the alias if the it was allocated inline, or 0 if there will be
+     * an asynchronous notification coming later. */
+    NodeAlias get_allocated_alias(NodeID destination_id, Executable *done);
 
     /** Releases a given alias. Sends out an AMR frame and puts the alias into
      * the reserved aliases queue. */
     void return_alias(NodeID id, NodeAlias alias);
 
+    /** Call from an alternate alias allocator. Marks that alias is reserved
+     * for the local interface (RID frame is just sent out). Adds the alias to
+     * the local alias cache and wakes up a flow that might be waiting for an
+     * alias.
+     * @param alias a reserved node alias. */
+    void add_allocated_alias(NodeAlias alias);
+
     /** If there is a pending alias allocation waiting for the timer to expire,
      * finishes it immediately. Needed in test destructors. */
     void TEST_finish_pending_allocation();
 
     /** Adds an allocated aliad to the reserved aliases queue.
         @param alias the next allocated alias to add.
     */
-    void TEST_add_allocated_alias(NodeAlias alias, bool repeat=false);
-    
+    void TEST_add_allocated_alias(NodeAlias alias);
```

Comment:
Q, should these TEST_XXX methods be guarded by a GTEST define so that they do not unintentionally get used at runtime?

Reply:
done